salt/utils/event.py

```diff
@@ -210,6 +217,9 @@ class SaltEvent(object):
     RAET compatible
     The base class used to manage salt events
     '''
+
+    SCHEMAS = []
+
     def __init__(
             self, node, sock_dir=None,
```

Comment:
```
if self.opts['schemas_enabled']:
    self.schemas = salt.loader.schemas(self.opts)
else:
    self.schemas = {}
```
so typically at class instantiation time instead of dynamically updating a class level list inside the `__init__`. That way you just have an assurance that the schemas are created and don't need to double check during runtime.

Reply:
I definitely like this but the reason for doing class level was because I didn't want to load schemas every time event object is created. This way a salt process doesn't reload schemas every time instance is created which during the life time of the process would be many times. wdyt ?